### PR TITLE
Replace CMake tool version display mechanism

### PR DESCRIPTION
**Pull request by gautesl**
Added a custom role `:tool-version:` that collects tool version information from pip requirement files and os-specific "tool-versions" files (including .yml files).

Example of use:
``:tool-version:`CMAKE_VERSION_LINUX` ``
``:tool-version:`SPHINX_VERSION` ``

Also removed the CMake tool version system and the corresponding file `versions.txt.in` and replaced it with the new `:tool-version:` custom role.

### CODE REVIEW

File: doc/_extensions/tool_versions.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+# Ordering matters, later entries will overwrite the earlier.
+PIP_REQUIREMENT_FILE_PATHS = [
+    ZEPHYR_SCRIPTS_BASE / "requirements-base.txt",
+    ZEPHYR_SCRIPTS_BASE / "requirements-doc.txt",
+    ZEPHYR_MCUBOOT_SCRIPTS_BASE / "requirements.txt",
+    NRF_SCRIPTS_BASE / "requirements-base.txt",
+    NRF_SCRIPTS_BASE / "requirements-doc.txt",
+    NRF_SCRIPTS_BASE / "requirements-build.txt",
+]
```
Comment (gmarull) [inline doc/_extensions/tool_versions.py:60]: these should come as extension config parameters

File: doc/_extensions/tool_versions.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            logger.error(f"Could not find tool version for '{text}'")
+            node = nodes.Text("")
+
+        return [node], []
+
+    return tool_version_role
+
+
+def setup(app: Sphinx):
+    app.add_role("tool-version", tool_version_replace())
```
Comment (gmarull) [inline doc/_extensions/tool_versions.py:136]: I'd use:

```suggestion
    app.add_role("ncs-tool-version", tool_version_replace())
```

File: doc/nrf/conf.py [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "sphinx_togglebutton",
     "sphinx_copybutton",
     "notfound.extension",
+    "tool_versions",
```
Comment (gmarull) [inline doc/nrf/conf.py:54]: name:
```suggestion
    "ncs_tool_versions",
```

**Review by gmarull — COMMENTED** [review]
lgtm, just a couple of suggestions

File: doc/_extensions/tool_versions.py (hunk 1 above)
Comment (gautesl) [reply doc/_extensions/tool_versions.py]: Agreed, thanks!

File: doc/_extensions/tool_versions.py (hunk 2 above)
Comment (gautesl) [reply doc/_extensions/tool_versions.py]: Fixed, thanks!

File: doc/_extensions/ncs_tool_versions.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    {TOOL_NAME}_VERSION_{OS}
+where OS is one of WIN10, LINUX or DARWIN
+
+For pip requirement files:
+    {TOOL_NAME}_VERSION
+
+Examples of use:
+- :tool-version:`CMAKE_VERSION_LINUX`
+- :tool-version:`SPHINX_VERSION`
+- :tool-version:`SPHINX_NCS_THEME_VERSION`
```
Comment (gmarull) [inline doc/_extensions/ncs_tool_versions.py:24]:
```suggestion
- :ncs-tool-version:`CMAKE_VERSION_LINUX`
- :ncs-tool-version:`SPHINX_VERSION`
- :ncs-tool-version:`SPHINX_NCS_THEME_VERSION`
```

File: doc/_extensions/ncs_tool_versions.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            logger.error(f"Could not find tool version for '{text}'")
+            node = nodes.Text("")
+
+        return [node], []
+
+    return tool_version_role
+
+
+def setup(app: Sphinx):
+    app.add_config_value("tool_version_file_paths", None, "env")
```
Comment (gmarull) [inline doc/_extensions/ncs_tool_versions.py:116]:
```suggestion
    app.add_config_value("ncs_tool_versions_host_deps", None, "env")
```

File: doc/_extensions/ncs_tool_versions.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    app.add_config_value("pip_requirement_file_paths", None, "env")
```
Comment (gmarull) [inline doc/_extensions/ncs_tool_versions.py:117]:
```suggestion
    app.add_config_value("ncs_tool_versions_python_deps", None, "env")
```

File: doc/nrf/getting_started/recommended_versions.rst [hunk 7]
```diff
@@ -117,21 +117,21 @@ Other versions might also work, but are not verified.
          * - Tool
            - Toolchain Manager version
          * - Zephyr SDK
-           - |zephyr_sdk_recommended_ver_win10|
+           - :tool-version:`ZEPHYR_SDK_VERSION_WIN10`
```
Comment (gmarull) [inline doc/nrf/getting_started/recommended_versions.rst:120]: needs update, also commit messages with `ncs-...`

Comment (gautesl) [reply doc/nrf/getting_started/recommended_versions.rst]: Should be changed everywhere now

**Review by gmarull — APPROVED** [review]
good job, happy to see the cmake part gone

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**